client/starwhale/core/job/model.py

```diff
@@ -1,456 +1,268 @@
-from __future__ import annotations
-
-import json
 import typing as t
-from abc import ABCMeta, abstractmethod
-from http import HTTPStatus
-from collections import defaultdict
-
-import jsonlines
-
-from starwhale.utils import load_yaml
-from starwhale.consts import HTTPMethod, DEFAULT_PAGE_IDX, DEFAULT_PAGE_SIZE
-from starwhale.base.uri import URI
-from starwhale.utils.fs import move_dir
-from starwhale.base.type import EvalTaskType, InstanceType, JobOperationType
-from starwhale.base.cloud import CloudRequestMixed
-from starwhale.utils.http import ignore_error
-from starwhale.utils.error import NotFoundError, NoSupportError
-from starwhale.utils.config import SWCliConfigMixed
-from starwhale.utils.process import check_call
-from starwhale.core.runtime.process import Process as RuntimeProcess
-
-from .store import JobStorage
-from .executor import EvalExecutor
-
-_device_id_map = {"cpu": 1, "gpu": 2}
-
-
-class Job(metaclass=ABCMeta):
-    def __init__(self, uri: URI) -> None:
-        self.uri = uri
-        self.name = uri.object.name
-        self.project_name = uri.project
-        self.sw_config = SWCliConfigMixed()
-
-    @classmethod
-    def create(
-        cls,
-        project_uri: URI,
-        model_uri: str,
-        dataset_uris: t.List[str],
-        runtime_uri: str,
-        name: str = "",
-        desc: str = "",
-        **kw: t.Any,
-    ) -> t.Tuple[bool, str]:
-        _cls = cls._get_job_cls(project_uri)
-        return _cls.create(
-            project_uri,
-            model_uri,
-            dataset_uris,
-            runtime_uri,
-            name,
-            desc,
-            **kw,
+from pathlib import Path
+from multiprocessing import Pipe, connection
+
+import yaml
+from loguru import logger
+
+from starwhale.core.job.loader import (
+    load_cls,
+    load_module,
+    get_func_from_module,
+    get_func_from_instance,
+)
+
+
+class Step:
+    def __init__(
+        self,
+        job_name: str,
+        step_name: str,
+        resources: str = "cpu=1",
```

Comment:
define a default resource const?

Reply:
use t.List[str] for resources is more friendly for the developer?

Reply:
done